drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -46,26 +47,21 @@
         implements DMNTypeService {
 
     @Override
-    public FactModelTuple retrieveType(Path path, String dmnPath) {
+    public FactModelTuple retrieveFactModelTuple(Path path, String dmnPath) throws WrongDMNTypeException {
         DMNModel dmnModel = getDMNModel(path, dmnPath);
         SortedMap<String, FactModelTree> visibleFacts = new TreeMap<>();
         SortedMap<String, FactModelTree> hiddenFacts = new TreeMap<>();
-
         ErrorHolder errorHolder = new ErrorHolder();
-
+        Map<String, List<String>> genericTypeInfoMap = new HashMap<>();
```

Comment:
Are you sure this map should be the same for all the `FactModelTree`s?
If I have understood correctly, this means that if you have two data objects with a list with the same field name (i.e. `Person.names` and `Book.names`) they will collapse in a single entry of the map

Reply:
@danielezonca 
I'll check - but I think there should be a "common" map - not sure about your case